ui/src/utils/index.ts

```diff
@@ -92,11 +92,11 @@ export const preprocessInput = (data: string, config: ILunaConfig) => {
     }
   }
 
-  if (data === '^C') {
-    data = '\x03';
+  if (data.includes('\u001b[200~') || data.includes('\u001b[201~')) {
+    return data.replace(/\u001b\[200~|\u001b\[201~/g, '');
+  } else {
+    return data;
   }
-
-  return data;
 };
 
 export const getOs = () => {
```

Comment:
The code looks fine to me. There are no differences between the two versions that I can see based on the knowledge cut-off of September 1st, 2021.

However, since there is still an issue with handling the `^C` (control-C) input being removed when it's included within other ANSI sequences for scrolling text (e.g., `\u001b[200~`, `\u001b[201~`), we need to include the solution from the second version provided:

```
if (
  !['\u001b', '\u007E'].includes(data.substring(0, 3))
) {
  // Check first three characters of input against expected sequence pattern
}
```